### ISSUE TITLE
Policy on including "ik", etc. variant forms.

### ISSUE THREAD

**Issue by jimmubreen** [JMdictProject/JMdictIssues#16]
In JMdict we allow for variant (often incorrect) forms to be included in entries when they are common enough to be encountered. We mark them with "ik", "iK", etc, tags to warn users that they are irregular or incorrect.  (In cases such as お待ち遠さま the incorrect お待ちどう様/お待ちどうさま is much more common than the correct form.)

Often we get into discussions as to whether a particular variant form is common enough to include. I would like to establish some policy guidelines which we can apply in these cases to avoid us having to justify and discuss the acceptance/rejection of a form.

As a possible policy, I suggest: "where a variant form occurs in the Google n-grams at least 5% as often as the correct form(s), or at least 2,000 times in total, whichever is less, then it will be accepted." Of course, we can still discuss other cases, but these guidelines can be applied fairly simply.

A couple of real/actual applications of this policy:

いらっしゃいませ	2662778
いらしゃいませ	30092 [ik]
いっらしゃいませ	4150 [ik]

The question has been raised as to whether いっらしゃいませ is common enough to be retained. Under this policy it would be kept as exceeds the 2,000 threshold.

いらっしゃい   7128308
いらしゃい   52227  [ik]
いっらしゃい    698

At present いっらしゃい is not in the entry, and under this policy it would not be included as it doesn't meet either threshold.

**Comment by ReneMalenfant** [comment]
That sounds okay to me.

I think we've generally been more stringent on using ik/iK when an entry appears in a kokugo (or gg5) than when it hasn't.  I.e., the default assumption we have been working on thus far might be something like, "if it's in a kokugo, assume it's **not** ik/iK by default; if it's not in a kokugo, assume it **is** ik/iK by default".  And if a kokugo in a note says that a particular way of writing something is incorrect, I think we've generally gone with what the kokugo suggested and marked it ik/iK regardless of how common it is.

For numerical inclusion guidelines (for headwords/readings that aren't in kokugos), both 5% and 2000 are arbitrary, but I suppose they would have to be.  5% sounds fair.  I don't really have a sense of what 2000 ngrams "feels like" in the real world, so I can't argue for or against that cutoff.

**Comment by robinjmdict** [comment]
While a threshold could be helpful in some cases, I don't think we should use absolute numbers. The n-gram counts for いっらしゃいませ are 0.16% of that for いらっしゃいませ. To me, that indicates that it's extremely uncommon and therefore not worth including. 

ありがとぅ occurs a whopping 118723 times in the n-grams, but ultimately it's only 0.12% of the regular form.

It's not difficult to find irregular forms that we would probably never add but nevertheless get many thousands of hits:
でしゅか	        49425
おはよぅ	        24059
ごじゃいます	22296
だめぇ	        9408
めっっっちゃ	6220
ありがとうう    5292
こんにちはあ	4686

Although we don't have to worry about physical space, I strongly feel we shouldn't clutter entries with forms like this. In most cases, a simple Google search will immediately point you to the regular form.

I think we'll have to continue to deal with irregular forms on a case-by-case basis because so many of them are simply not suitable for a dictionary entry despite being very common. For the less wacky forms, a 5% threshold doesn't sound unreasonable. And I agree with Rene that forms marked as irregular in the kokugos should be included regardless of how common they are.

**Comment by jimmubreen** [comment]
Where I am coming from in this matter is partly influenced by being a developer of dictionary applications. I want to help people find their way to the entry for, or related to. the a term they've encountered. [Years ago a correspondent pointed out they couldn't find "たべません" in WWWJDIC. I explained that it was "just" the plain past tense of 食べる, and he responded "well I don't necessarily know that", which in a way is a fair point. So I added to the base dictionary used in WWWJDIC a mass of kana-only inflected forms of common verbs. Now if you look up たべません you get a redirection to 食べる.]
Part of my interest in having good coverage of common but not really correct forms is to be helpful to people who don't/can't necessarily work out the dictionary form of what they are seeking. A second interest is to help text-glossing apps. Forms such as ありがとうう and こんにちはあ are not such a problem as apps such as WWWJDIC will show enough of a partial match to connect to the correct entry. Things like いっらしゃいませ are more of a problem as they don't match enough of the correct form for a partial match to be much use, and at 4k+ occurrences it's possibly common enough for a beginner to encounter and not realise it's a broken version of いらっしゃいませ. I can always add to to the extras file I use for text-glossing, and I could have a "B-list" of mistaken terms, but I'm still interested in a reasonable "ik" coverage.
BTW, でしゅか is an interesting one. Unidic has でしゅ as a variant of です - I guess the NINJAL people have encountered it enough to make it wort handling.
